src/rpc/mining.cpp

```diff
@@ -745,12 +746,13 @@ static UniValue submitblock(const JSONRPCRequest& request)
         }
     }
 
-    bool new_block;
     submitblock_StateCatcher sc(block.GetHash());
     RegisterValidationInterface(&sc);
-    bool accepted = ProcessNewBlock(Params(), blockptr, /* fForceProcessing */ true, /* fNewBlock */ &new_block);
+    bool new_block = ProcessNewBlock(Params(), blockptr, /* fForceProcessing */ true).get();
     UnregisterValidationInterface(&sc);
-    if (!new_block && accepted) {
+    LOCK(cs_main);
+    const CBlockIndex* pindex = LookupBlockIndex(blockptr->GetHash());
+    if (!new_block && pindex && pindex->IsValid()) {
```

Comment:
In commit "Make ProcessNewBlock return a future instead of an immediate bool" (79e6e446dd0c8a3f53759a9792388f28e1053bfb)

This seems fine, but just to make sure I understand correctly, there is a change in behavior here? Previously `accepted` would only be true if `CheckBlock` and `AcceptBlock` and `ActivateBestChain` all succeeded but this can be true even they fail?